### PR TITLE
[MXNET-825] Fix CGAN R Example with MNIST dataset

### PR DESCRIPTION
**Pull request by anirudhacharya**
## Description ##
Fix the example for generating MNIST digit images - add context for optimizer update, update metrics, clean up some code.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

## Comments ##
- @hetong007 @ankkhedia

### CODE REVIEW

File: example/gan/CGAN_mnist_R/CGAN_train.R [hunk 1]
```diff
@@ -167,6 +173,7 @@ for (iteration in 1:2400) {
   }
 }
 
+dir.create(file.path(".", "models"))
```
Comment (hetong007) [inline example/gan/CGAN_mnist_R/CGAN_train.R]: Call `dir.exists` first.

File: example/gan/CGAN_mnist_R/CGAN_train.R [hunk 2]
```diff
@@ -46,11 +47,11 @@ exec_D<- mx.simple.bind(symbol = D_sym, data=data_shape_D, digit=digit_shape_D,
 ### initialize parameters - To Do - personalise each layer
 initializer<- mx.init.Xavier(rnd_type = "gaussian", factor_type = "avg", magnitude = 3)
 
-arg_param_ini_G<- mx.init.create(initializer = initializer, shape.array = mx.symbol.infer.shape(G_sym, data=data_shape_G)$arg.shapes, ctx = mx.cpu())
-aux_param_ini_G<- mx.init.create(initializer = initializer, shape.array = mx.symbol.infer.shape(G_sym, data=data_shape_G)$aux.shapes, ctx = mx.cpu())
+arg_param_ini_G<- mx.init.create(initializer = initializer, shape.array = mx.symbol.infer.shape(G_sym, data=data_shape_G)$arg.shapes, ctx = devices)
```
Comment (hetong007) [inline example/gan/CGAN_mnist_R/CGAN_train.R]: 1. This line is too long
2. leave a space before `<-`

Please check the entire file with the above two comments.

File: example/gan/CGAN_mnist_R/CGAN_train.R (hunk 1 above)
Comment (anirudhacharya) [reply example/gan/CGAN_mnist_R/CGAN_train.R]: if the folder already exists, it will just throw a warning message.

Comment (hetong007) [reply example/gan/CGAN_mnist_R/CGAN_train.R]: Yes, but it's still better to have no warning in a ready-to-run example.

**Comment by ankkhedia** [comment]
Looks good! Could you also in some comments add the sequence in which files are run and also to make sure 'iterator.R' is in the same folder before this line: https://github.com/apache/incubator-mxnet/blob/fdb28391be2876277011b1d485abe9a7db13b88e/example/gan/CGAN_mnist_R/CGAN_mnist_setup.R#L23

**Comment by anirudhacharya** [comment]
@ankkhedia since the other file is named setup I think we do not have to specifically mention the order in which the files have to be run.

**Comment by anirudhacharya** [comment]
@hetong007 ping for review.

File: example/gan/CGAN_mnist_R/CGAN_train.R [hunk 3]
```diff
@@ -64,32 +80,33 @@ input_names_D <- mxnet:::mx.model.check.arguments(D_sym)
 
 ###################################################
 #initialize optimizers
-optimizer_G<-mx.opt.create(name = "adadelta",
+optimizer_G <- mx.opt.create(name = "adadelta",
                            rho=0.92,
```
Comment (hetong007) [inline example/gan/CGAN_mnist_R/CGAN_train.R]: I still see some inconsistent format here and there. Please consider using tools like [formatR](https://yihui.name/formatr/) to clean up the style.

File: example/gan/CGAN_mnist_R/CGAN_train.R [hunk 4]
```diff
@@ -15,9 +15,116 @@
...
+train_data <- t(train_data/255 * 2 - 1)
+train_label <- as.integer(train[, 1])
+
+dim(train_data) <- c(28, 28, 1, ncol(train_data))
+
+### Model parameters
+random_dim <- 96
+gen_features <- 96
+dis_features <- 32
+image_depth = 1
```
Comment (hetong007) [inline example/gan/CGAN_mnist_R/CGAN_train.R]: `formatR` has option to check assign operator.

**Comment by sandeep-krishnamurthy** [comment]
@anirudhacharya @hetong007 - What is the current status here?
Thanks for your contributions!

**Pull request merged**